app/facades/bill_search_facade.rb

```diff
@@ -2,25 +2,66 @@ class BillSearchFacade
   def initialize(params)
     @params = params
   end
-  
+
   def bills
     bill_search = BillService.new(@params).bills_by_query
     bill_objects = []
-    bill_search[:searchresult].drop(1).map do |bill_data|
+    bill_search[:searchresult].drop(1).each do |bill_data|
       bill_hash = {id: bill_data.first.to_s.to_i,
         attributes: bill_data.last }
-        
-        bill_objects << ApiBill.new(bill_hash)
-      end
+        bill_object = ApiBill.new(bill_hash)
+        bill_objects << bill_object
       
+      create_update_bills(bill_data.last)
+    end
       bill_objects
+  end
+
+  def bill
+    bill_search = BillService.new(@params).bills_by_id
+    bill_hash = { id: bill_search[:bill][:bill_id],
+      attributes: bill_search[:bill]
+    }
+    bill_object = ApiBill.new(bill_hash)
+    bill_object
+  end
+
+  def bill_sponsors 
+
+    bill_search = BillService.new(@params).bills_by_id
+
+    bill_search[:bill][:sponsors].map do |sponsor|
+      bill_hash = { first_name: sponsor[:first_name],
+                  last_name: sponsor[:last_name] }
+                  
+      ApiMember.new(bill_hash)
     end
-    
-    def bill
-      bill_search = BillService.new(@params).bills_by_id
-      bill_hash = { id: bill_search[:bill][:bill_id],
-        attributes: bill_search[:bill]
-      }
-      ApiBill.new(bill_hash)
   end
-end
\ No newline at end of file
+  
+  
+  private
+  
+  def create_update_bills(bill_data)
+    bill1 = Bill.find_or_create_by(bill_id: bill_data[:bill_id]) do |b|
+      b.bill_number = bill_data[:bill_number]
+      b.text_url = bill_data[:text_url]
+      b.last_action_date = bill_data[:last_action_date]
+      b.last_action = bill_data[:last_action]
+      b.title = bill_data[:title]
+      b.state = bill_data[:state]
+    end
+
+    unless bill1.new_record?
+      bill1.update(
+        bill_number: bill_data[:bill_number],
+        text_url: bill_data[:text_url],
+        last_action_date: bill_data[:last_action_date],
+        last_action: bill_data[:last_action],
+        title: bill_data[:title],
+        state: bill_data[:state]
+      )
```

Comment:
curious about this last method after the unless bill is new_record? to update the bill... why are we updating the bill with the same information we created the bill with? should this update portion be updating the bill from a getBill action with the status, description, sponsors, texts etc? If so, we should also be calling this method in the `def bill` method above right?